commerce/domain/tax/cart/calculator/classes/TaxCartCalculatorSample.cls

```diff
@@ -17,8 +17,9 @@ public class TaxCartCalculatorSample extends CartExtension.TaxCartCalculator {
       // Clean up CVO based on tax. When new tax calculator request comes, we need to clean up
       // previous CVOs as they have been previously handled by the Cart Calculate API.
       CartExtension.CartValidationOutputList cartValidationOutputCollection = cart.getCartValidationOutputs();
-      for (Integer i = (cartValidationOutputCollection.size() - 1); i >= 0; i--) {
-        CartExtension.CartValidationOutput cvo = cartValidationOutputCollection.get(i);
+      Iterator<CartExtension.CartValidationOutput> cartValidationOutputCollectionIterator = cartValidationOutputCollection.iterator();
+      while (cartValidationOutputCollectionIterator.hasNext()) {
+        CartExtension.CartValidationOutput cvo = cartValidationOutputCollectionIterator.next();
         if (cvo.getType() == CartExtension.CartValidationOutputTypeEnum.TAXES) {
           cartValidationOutputCollection.remove(cvo);
```

Comment:
In java removing from collection while iteration would result into `ConcurrentModificationException`. Could you please verify if such behavior is applied in Apex as well?

Reply:
From a legibility perspective, this suggestion feels better. Doing the backward loop to avoid one-off errors brought about by element removal during a foreach loop seems overengineered, but maybe it's a limitation of the language? Does the hasNext approach protect against one-offs? I'm no Apex expert, but in many languages foreach and iterator can land equivalent in terms of execution.

Reply:
When i tested, this iterator working as expected without throwing `ConcurrentModificationException` error but for better understanding and more readability, i refactored the code.